test/unit/Git/CommitFileViaConsoleTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\AutomaticReleases\Test\Unit\Git;
+
+use Laminas\AutomaticReleases\Git\CommitFileViaConsole;
+use Laminas\AutomaticReleases\Git\Value\BranchName;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Process\Process;
+use Webmozart\Assert\Assert;
+
+use function file_put_contents;
+use function Safe\tempnam;
+use function sprintf;
+use function sys_get_temp_dir;
+use function unlink;
+
+final class CommitFileViaConsoleTest extends TestCase
+{
+    /** @psalm-var non-empty-string */
+    private string $checkout;
+
+    public function setUp(): void
+    {
+        $checkout = tempnam(sys_get_temp_dir(), 'CommitFileViaConsoleTestCheckout');
+        Assert::notEmpty($checkout);
+
+        $this->checkout = $checkout;
+        unlink($this->checkout);
+
+        (new Process(['git', 'init', $this->checkout]))
+            ->mustRun();
+
+        (new Process(
+            ['git', 'symbolic-ref', 'HEAD', 'refs/heads/1.0.x'],
```

Comment:
Fancy trick! This is basically a `git switch`, but to a new branch?

Reply:
Yep! Discovered it when I was trying to figure out how to test this yesterday. What it does is change the symbolic name of the HEAD to what you provide; it's a way to rename the current branch, essentially.